server.js

```diff
@@ -1,34 +1,110 @@
+import listEndpoints from 'express-list-endpoints'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
+import booksData from './data/books.json'
+
+const port = process.env.PORT || 8000
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
-// Start defining your routes here
+// route to APIs first page listing all possible endpoints
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
 })
 
-// Start the server
-app.listen(port, () => {
-  // eslint-disable-next-line
-  console.log(`Server running on http://localhost:${port}`)
+// endpoint to get all books
+app.get('/books', (req, res) => {
+  res.json({ data: booksData })
+})
+
+// endpoint with search path and query params
+app.get('/books/search', (req, res) => {
+  // destructuring query params 
+  const { 
+    author, 
+    title, 
+    highToLow, 
+    lowToHigh, 
+    longStory, 
+    shortStory, 
+    topTwenty, 
+    topFiFtyRatingsCount, 
+    isbn 
+  } = req.query 
+
+  let queriedBooks = booksData 
+
+  // Query to find author.
+  if (author) {
+    queriedBooks = queriedBooks
+      .filter((book) => book.authors.toLowerCase().includes(author.toLowerCase()))
+  } 
+  // query for title
+  if (title) {
+    queriedBooks = queriedBooks
+      .filter((book) => book.title.toString().toLowerCase().includes(title.toLowerCase()))
+  }
+  // query to sort the books from high to low in average rating. 
+  if (highToLow) {
+    queriedBooks = queriedBooks
+      .filter((book) => book.average_rating)
+      .sort((a, b) => b.average_rating - a.average_rating)
+  }
+  // query to sort the books from low to high in average rating.
+  if (lowToHigh) {
+    queriedBooks = queriedBooks
+      .filter((book) => book.average_rating)
+      .sort((a, b) => a.average_rating - b.average_rating)
+  }
+  // query to show books over or equal to 400 pages
+  if (longStory) {
+    queriedBooks = queriedBooks
+      .filter((book) => book.num_pages >= 400)
+  }
+  // query to show books under or equal to 400 pages
+  if (shortStory) {
+    queriedBooks = queriedBooks
+      .filter((book) => book.num_pages <= +400)
+  }
+  // listing the top 20 highest rated books, in descending order
+  if (topTwenty) { 
+    queriedBooks = queriedBooks
+      .filter((book) => book.average_rating)
+      .sort((a, b) => b.average_rating - a.average_rating)
```

Comment:
You repeating it a lot, so it would be good to extract it into a separate function and call it when needed 🤓